### ISSUE TITLE
How to load resnet models with the given checkpoints?

### ISSUE THREAD

**Issue by HashmatShadab** [microsoft/robust-models-transfer#14]
Can you please provide the script to load resnet model with the given checkpoints in the repo?

**Comment by Hadisalman** [comment]
Hi @HashmatShadab. Are you sure you are normalizing the inputs? The loaded model should have a normalization layer, but by doing `model = resnet.model
` you got rid of that layer, so you have to manually normalize. 

See the implementation of the [AttackerModel](https://github.com/MadryLab/robustness/blob/4033befe273b29f7b6dc36c30aa40696ed8fae96/robustness/attacker.py#L253) from our robustness lib, which is the model that we save in these checkpoints.

**Comment by HashmatShadab** [comment]
I was manually normalizing it by :+1: 
transform = transforms.Compose([
transforms.Resize((224, 224)),
transforms.ToTensor(),
transforms.Normalize(mean=[0.485, 0.456, 0.406],
                   std=[0.229, 0.224, 0.225])])

**Comment by HashmatShadab** [comment]
batch_size = 10
device = "cuda" if torch.cuda.is_available() else "cpu"

imagenet_ds = ImageNet('data/pathl')
model , _ = model_utils.make_and_restore_model(arch="resnet18", dataset=imagenet_ds,
                                                          resume_path='pretrained-models/resnet18_l2_eps0.ckpt', parallel=False)
model.eval()
model.to(device)
transform = transforms.Compose([
                                        transforms.Resize((248, 248)),
                                        transforms.CenterCrop(224),
                                        transforms.ToTensor(), ]
                                        )

correct = 0
total = 0
count = 0

for i, (images, labels) in enumerate(data_loader):
    images, labels = images.to(device), labels.to(device)
    preds = model(images, with_image=False)
    _, pre = torch.max(preds, 1)
    correct += torch.sum(pre == labels)
    total += images.shape[0]
    print(total, end="\r")
print(f"The Model accuracy is {correct / total}")
The Model accuracy is 0.6879199743270874

Now I am achieving a better accuracy but still less than  69.79%. Please let me know is there anything I am missing?

**Comment by HashmatShadab** [comment]
Compose(
    Resize(size=256, interpolation=bilinear)
    CenterCrop(size=(224, 224))
    ToTensor()
)
I just found out that this transformation is used during evaluation. Using this transformation the accuracy increased to 69.1%, but still less than the mentioned 69.79%

**Comment by HashmatShadab** [comment]
I was able to reproduce the results by loading the data using :
from robustness.datasets import ImageNet
imagenet_ds = ImageNet('path/to/data')
train_loader, test_loader = imagenet_ds.make_loaders(batch_size=10, workers=0)

However, the same transformations are applied in the make_loader as I had done manually. Please let me know what can be the issue in getting slightly lower accuracy when manually adding transformations?
